src/interfaces/pyo3/py_cover_crypt.rs

```diff
@@ -0,0 +1,404 @@
+use cosmian_crypto_core::{
+    bytes_ser_de::{Deserializer, Serializable, Serializer},
+    KeyTrait,
+};
+use pyo3::{exceptions::PyException, exceptions::PyTypeError, prelude::*, types::PyType};
+
+use crate::{
+    api::CoverCrypt as CoverCryptRust,
+    interfaces::pyo3::py_abe_policy::Policy,
+    interfaces::statics::{
+        CoverCryptX25519Aes256, EncryptedHeader, MasterSecretKey as MasterSecretKeyRust,
+        PublicKey as PublicKeyRust, SymmetricKey as SymmetricKeyRust,
+        UserSecretKey as UserSecretKeyRust,
+    },
+};
+
+// Pyo3 doc on classes
+// https://pyo3.rs/v0.16.2/class.html
+
+#[pyclass]
+pub struct MasterSecretKey {
+    inner: MasterSecretKeyRust,
+}
+
+#[pymethods]
+impl MasterSecretKey {
+    /// Convert to bytes for sending/saving the key
+    pub fn to_bytes(&self) -> PyResult<Vec<u8>> {
+        match self.inner.try_to_bytes() {
+            Ok(res) => Ok(res),
+            Err(e) => Err(PyException::new_err(e.to_string())),
+        }
+    }
+
+    /// Read key from bytes
+    #[classmethod]
+    pub fn from_bytes(_cls: &PyType, key_bytes: Vec<u8>) -> PyResult<Self> {
+        match MasterSecretKeyRust::try_from_bytes(&key_bytes) {
+            Ok(key) => Ok(MasterSecretKey { inner: key }),
+            Err(e) => Err(PyException::new_err(e.to_string())),
+        }
+    }
+}
+
+#[pyclass]
+pub struct PublicKey {
+    inner: PublicKeyRust,
+}
+
+#[pymethods]
+impl PublicKey {
+    /// Convert to bytes for sending/saving the key
+    pub fn to_bytes(&self) -> PyResult<Vec<u8>> {
+        match self.inner.try_to_bytes() {
+            Ok(res) => Ok(res),
+            Err(e) => Err(PyException::new_err(e.to_string())),
+        }
+    }
+
+    /// Read key from bytes
+    #[classmethod]
+    pub fn from_bytes(_cls: &PyType, key_bytes: Vec<u8>) -> PyResult<Self> {
+        match PublicKeyRust::try_from_bytes(&key_bytes) {
+            Ok(key) => Ok(PublicKey { inner: key }),
+            Err(e) => Err(PyException::new_err(e.to_string())),
+        }
+    }
+}
+
+#[pyclass]
+pub struct UserSecretKey {
+    inner: UserSecretKeyRust,
+}
+
+#[pymethods]
+impl UserSecretKey {
+    /// Convert to bytes for sending/saving the key
+    pub fn to_bytes(&self) -> PyResult<Vec<u8>> {
+        match self.inner.try_to_bytes() {
+            Ok(res) => Ok(res),
+            Err(e) => Err(PyException::new_err(e.to_string())),
+        }
+    }
+
+    /// Read key from bytes
+    #[classmethod]
+    pub fn from_bytes(_cls: &PyType, key_bytes: Vec<u8>) -> PyResult<Self> {
+        match UserSecretKeyRust::try_from_bytes(&key_bytes) {
+            Ok(key) => Ok(UserSecretKey { inner: key }),
+            Err(e) => Err(PyException::new_err(e.to_string())),
+        }
+    }
+}
+
+#[pyclass]
+pub struct SymmetricKey {
+    inner: SymmetricKeyRust,
+}
+
+#[pymethods]
+impl SymmetricKey {
+    /// Convert to bytes for sending/saving the key
+    pub fn to_bytes(&self) -> PyResult<Vec<u8>> {
+        Ok(self.inner.to_bytes().to_vec())
```

Comment:
This avoids one copy:
```suggestion
        Ok(self.inner.as_bytes().to_vec())
```
(+ add `use cosmian_crypto_core::symmetric_crypto::SymKey`)